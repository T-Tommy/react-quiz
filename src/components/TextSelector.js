import React from 'react';
import PropTypes from 'prop-types';

function TextSelector({ text, handleChange }) {
  return (
    <input
      type="text"
      name="text"
      value={ text }
      onInput={ handleChange }
    />
  );
}

TextSelector.propTypes = {
  text: PropTypes.string.isRequired,
  handleChange: PropTypes.func.isRequired
};

export default TextSelector;
