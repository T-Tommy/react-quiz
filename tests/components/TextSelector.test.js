import React from 'react';
import { shallow } from 'enzyme';
import TextSelector from '../../src/components/TextSelector';

describe('TextSelector component', () => {
  it('renders the TextSelector', () => {
    const wrapper = shallow(<TextSelector handleChange={ jest.fn } text="test" />);
    expect(wrapper).toMatchSnapshot();
  });
});
