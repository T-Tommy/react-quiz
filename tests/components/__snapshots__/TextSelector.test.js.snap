// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextSelector component renders the TextSelector 1`] = `
<input
  name="text"
  onInput={[Function]}
  type="text"
  value="test"
/>
`;
